src/os/bluestore/BlueStore.cc

```diff
@@ -10393,8 +10393,10 @@ int BlueStore::_do_remove(
           }
         }
 
-	h->extent_map.dirty_range(b_start, b_end - b_start);
-	txc->write_onode(h);
+        if (b_end > b_start) {
```

Comment:
I think this should be an assert?  blobs in unshared_blobs are definitely modified and if we don't encounter references to them in the extent_map then there is a bug/problem (we won't dirty and persist the change)